### DISCUSSION TITLE
Why not adopt Calamares as graphical solution instead of relying on terminal ?

### DISCUSSION THREAD

**Discussion by medmedin2014** [archlinux/archinstall#2160] [Q&A]
The command line gives a lot of problems and can lead the user to dangerous situations especially when choosing partitions, so why is Calamares not preferred as an alternative ? Many Arch based distributions already use it, and it works fine, even some unresolved issues with archinstall like installing on single ext4 partition is not present with Calamares.

**Comment by Torxed** [comment]
Thank you for asking, it's a pretty fun question and I had to think a bit about it. Read the following with a cheerful tone - not an attacking tone as I genuinely had fun answering it - but I'm going to challenge your perspective and that will come across as harsh I believe.

I would like you to expand a few things which to me are subjective and non-descriptive as it's currently written. For instance the sentence:

> The command line gives a lot of problems

What kind of problems are you referring to? Using a terminal based UI we actually get some advantages. One being that it will work on almost any hardware *(headless, servers, IoT)* as well as being easy to make accessible, for instance to visually impaired. 

> can lead the user to dangerous situations

In what way can a TUI lead to more dangerous situations than GUI based options? Because the user is not used to the terminal? If you're specifically referring to partitioning, then there's is no harm in using gparted and then setting the rest up in archinstall - actually that is a pretty nice way of doing it if you really need a graphical partitioning tool.. hard to beat :)

> Many Arch based distributions already use it

They're distributions and as such are free to do whatever they choose, I've never pushed `archinstall` on them neither do I expect others to push things on me. The distro's *probably* *(I say probably because I've never really talked in depth with any of the distro creators)* chose a finished "installer" like Anaconda or even Calamares because they wanted to focus on the distro experience - not the installer. I however love Arch in itself and don't need a pre-packaged and pre-maintained distro abstraction so I chose to focus my spare time on writing an installer tailored specifically to the Arch platform. With that said, feedback is welcome and I do consider every option. However, a heavily C++ based application targeted to generic environments is not something I'm considering. Their release cadence appears to have dimmed down a bit and their issues appears to be in line with what we're struggling with too - so to be honest I'd rather keep `archinstall` which aims to be a library just like calamares.. the difference being I can vouch for the source code of `archinstall` as I've personally vetted and looked over the entire source code from beginning to my eventual demise.

If I knew back in 2013 that this library *(and it's installer that you probably know it as)* would get *this* much traction, I would for sure re-do some things.. Not that `dataclass` even existed, let alone Pydantic.. But I'd separate some of the duties a bit more and make more use of classes and instances for objects and types.

> issues with archinstall like installing on single ext4 partition

I'm assuming you're referring to MBR partitioning with a single partition for root and letting GRUB deal with booting everything from that single ext4 partition?

tl;dr: I like the fact that we're running Python, it's quite easy to maintain and grasp. I like that `archinstall` is a library - just like Calamares kinda is. I like that I have full audit of the source code in `archinstall` which means I can trust it to do only what I know it will do. It's pretty small considering. Very easy to customize if you're familiar with Python *(although some parts have gotten crazy abstractions, which I'm looking over)*. We have a great community with 174 contributors, used by quite a few people and as long as there's hope and people cheer on - I'll personally keep improving this tool and I love to see the awesome work by the other contributors who last year have done more coding than me I recon.
